### ISSUE TITLE
More comments and fewer shortcuts in scripts to show process better

### ISSUE THREAD

**Issue by Ace-Dragon** [godotengine/godot#1617]
The request would especially apply to demos geared toward beginners, the demo files right now make quite a use of multiple functions being done in one line similar to what you can do in python, currently it's something like...

*var A = (get_functionA().functionB().functionC())*
____________________________________________

Possibly, one can make it more understandable for people to make it more readable like doing...

*var A = get_functionA()*
*var B = A.functionB()*
*var C = B.functionC()*
____________________________________________

What also confuses newcomers even is that these scripts often have very few comments. There should be a requirement that everything in the demo pack have a decent number of comments in the code that is telling people what the functions are doing, so it would be...

*var A = get_functionA() #what it does*
*var B = A.functionB() #what it does*
*var C = B.functionC() #what it does*
______________________________________________

So, in conclusion, would this be a good new convention to adopt for the demos (since they're supposed to be for learning purposes)?

**Comment by Wend1go** [comment]
I second this.
Best example is the "Navmesh demo". Lots of code for calculating the path with only a single comment.
There should also be a multiline comment at the top of each script that summarises what the code actually does. A license would also be nice.

**Comment by akien-mga** [comment]
Adding more comments is probably a good idea. I'm not sure about making a possibility to split all commands as you described above, sometimes it makes sense, but sometimes it's IMO unecessarily detailed.

> A license would also be nice.

All of the repo (apart from the external libs) is under the MIT license.

**Comment by akien-mga** [comment]
> > A license would also be nice.

> All of the repo (apart from the external libs) is under the MIT license.

I've added an explicit license now btw.

**Comment by djrm** [comment]
movable to demos repo

**Comment by bojidar-bg** [comment]
Done.
